hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -243,19 +237,21 @@ case class HoodieFileIndex(spark: SparkSession,
       prunePartitionsAndGetFileSlices(dataFilters, partitionFilters)
     hasPushedDownPartitionPredicates = true
 
+    val prunedPartitionsAndFileSlicesWithLogs = prunedPartitionsAndFileSlices.map(s => (s._1, s._2.map(f => f.withLogFiles(includeLogFiles))))
```

Comment:
Let's add the transformation in `prunePartitionsAndGetFileSlices` instead of here
```
    (prunedPartitionsTuple._1, getInputFileSlices(prunedPartitionsTuple._2: _*).asScala.map(
      { case (partition, fileSlices) => (Option.apply(partition), fileSlices.asScala.map(f => f.withLogFiles(includeLogFiles)).toSeq) }).toSeq)
```